packages/edit-site/src/components/resizable-frame/index.js

```diff
@@ -33,108 +33,124 @@ const HANDLE_STYLES_OVERRIDE = {
 };
 
 const MIN_FRAME_SIZE = 340;
-const lerp = ( a, b, amount ) => {
-	return a + ( b - a ) * amount;
-};
 
-function ResizableFrame( { isFull, children } ) {
+function calculateNewHeight( width, initialAspectRatio ) {
+	const lerp = ( a, b, amount ) => {
+		return a + ( b - a ) * amount;
+	};
+
+	// Calculate the intermediate aspect ratio based on the current width.
+	const lerpFactor =
+		1 -
+		Math.max(
+			0,
+			Math.min(
+				1,
+				( width - MIN_FRAME_SIZE ) / ( 1300 - MIN_FRAME_SIZE )
+			)
+		);
+
+	// Calculate the height based on the intermediate aspect ratio
+	// ensuring the frame arrives at a 9:19.5 aspect ratio.
+	const intermediateAspectRatio = lerp(
+		initialAspectRatio,
+		9 / 19.5,
+		lerpFactor
+	);
+
+	return width / intermediateAspectRatio;
+}
+
+function ResizableFrame( { isFullWidth, children } ) {
 	const [ frameSize, setFrameSize ] = useState( {
 		width: '100%',
 		height: '100%',
 	} );
+	const [ startingWidth, setStartingWidth ] = useState();
 	const [ isResizing, setIsResizing ] = useState( false );
 	const [ isHovering, setIsHovering ] = useState( false );
 	const [ isOversized, setIsOversized ] = useState( false );
 	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
 	const initialAspectRatioRef = useRef( null );
 	const initialComputedWidthRef = useRef( null );
 	const FRAME_TRANSITION = { type: 'tween', duration: isResizing ? 0 : 0.5 };
+	const frameRef = useRef( null );
+
+	// Remember frame dimensions on initial render.
+	useEffect( () => {
+		const { offsetWidth, offsetHeight } = frameRef.current.resizable;
+		initialComputedWidthRef.current = offsetWidth;
+		initialAspectRatioRef.current = offsetWidth / offsetHeight;
+	}, [] );
+
+	const handleResizeStart = ( _event, _direction, ref ) => {
+		// Remember the starting width so we don't have to get `ref.offsetWidth` on
+		// every resize event thereafter, which will cause layout thrashing.
+		setStartingWidth( ref.offsetWidth );
+		setIsResizing( true );
+	};
 
 	// Calculate the frame size based on the window width as its resized.
-	const handleResize = ( event, direction, ref ) => {
-		const updatedWidth = ref.offsetWidth;
-		const updatedHeight = ref.offsetHeight;
+	const handleResize = ( _event, _direction, _ref, delta ) => {
+		const updatedWidth = startingWidth + delta.width;
+
+		setIsOversized( updatedWidth > initialComputedWidthRef.current );
+
+		setFrameSize( {
+			width: updatedWidth,
+			height: isOversized
+				? '100%'
+				: calculateNewHeight(
+						updatedWidth,
+						initialAspectRatioRef.current
+				  ),
+		} );
+	};
 
-		if ( initialComputedWidthRef.current === null ) {
-			initialComputedWidthRef.current = updatedWidth;
-		}
+	// Make the frame full screen when the user resizes it to the left.
```

Comment:
I'm not sure this comment is super clear — maybe something like "Based on the frame's final size, maybe snap the frame back to its initial position or switch to edit mode" ?

Reply:
I'm going to remove this comment since the logic is already commented inline 🧹